### PR TITLE
✨ PROS 4: Move device enum out of apix and motor funcs to apix

### PR DESCRIPTION
**Pull request by aberiggs**
#### Summary:
Created C version of device.hpp and device.cpp
- Created get_installed_type() for device, taking a port and returning v5_device_type_e_t (the type that is plugged into that port). 

Moved the enum v5_device_type_e_t from apix.h to device.h

Moved motor_set_reversed() and motor_is_reversed() from motors.h to apix.h

#### Motivation:
We're wanting to use features utilizing v5_device_type_e_t more and are updating how the device class works so it makes sense to move the enum to a C version of the device base class.

#### Test Plan:
To be determined

### CODE REVIEW

File: src/devices/vdml_device.c [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+v5_device_e_t get_plugged_type(uint8_t port) {
+    if (!port_mutex_take(port - 1)) {                            
+		errno = EACCES; 
+		return E_DEVICE_UNDEFINED;                                                                           
+	}
+	v5_device_e_t type = registry_get_plugged_type(port - 1);
+	
+	return_port(port - 1, type);
+}
```
Comment (WillXuCodes) [inline src/devices/vdml_device.c:29]:
```suggestion
}

```
Add newline back please

File: include/pros/device.h [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	E_DEVICE_GPS = 20,
+	E_DEVICE_SERIAL = 129,
+	E_DEVICE_GENERIC __attribute__((deprecated("use E_DEVICE_SERIAL instead"))) = E_DEVICE_SERIAL,
+	E_DEVICE_UNDEFINED = 255
+} v5_device_e_t;
+
+/**
+	 * Gets the type of device on given port.
+	 *
+	 * \return The device type as an enum.
```
Comment (WillXuCodes) [inline include/pros/device.h:53]:
```suggestion
* Gets the type of device on given port.
*
* \return The device type as an enum.
```

fix indentation

File: src/devices/vdml_motors.c [hunk 3]
```diff
@@ -17,6 +17,7 @@
 #include "kapi.h"
 #include "pros/colors.h"
 #include "pros/motors.h"
+#include "pros/apix.h"
```
Comment (WillXuCodes) [inline src/devices/vdml_motors.c:20]: why is this here? is this necessary?

**Review by WillXuCodes — CHANGES_REQUESTED** [review]

File: src/devices/vdml_motors.c (hunk 3 above)
Comment (aberiggs) [reply src/devices/vdml_motors.c:20]: Since I was told to move the prototypes of 2 of the functions within motors.h into apix.h, so I thought it made sense to import the header into vdml_motors.c since those 2 functions are now prototyped within apix.h. It appears to compile without, so if you'd rather me remove I can do that.

Comment (WillXuCodes) [reply src/devices/vdml_motors.c:20]: Got it! Srry forgot that was part of it lol

**Review by WillXuCodes — CHANGES_REQUESTED** [review]
Forgot to mention but device.h should be put under api.h

File: Makefile
Comment (noam987) [inline Makefile:1]: Can you make sure we get a newline at the end of this file

**Review by noam987 — APPROVED** [review]
LGTM except for the newline at the end of the make file missing

File: Makefile
Comment (aberiggs) [reply Makefile:1]: Oh shoot my bad, I'll do that right now.

**Review by WillXuCodes — APPROVED** [review]
LGTM
